pageserver/src/http/routes.rs

```diff
@@ -2357,8 +2357,9 @@ async fn get_utilization(
     // regenerate at most 1Hz to allow polling at any rate.
     if !still_valid {
         let path = state.conf.tenants_path();
-        let doc = crate::utilization::regenerate(path.as_std_path())
-            .map_err(ApiError::InternalServerError)?;
+        let doc =
+            crate::utilization::regenerate(state.conf, path.as_std_path(), &state.tenant_manager)
+                .map_err(ApiError::InternalServerError)?;
```

Comment:
I'm wondering if it makes sense to decouple the score regeneration from handling the request (now that regeneration is more involved).

Reply:
It would make sense to do this in the background, but we don't currently have a background housekeeping loop at the TenantManager scope, not sure it's worth introducing that infrastructure just for this.

Reply:
Let's see how it goes. Can always move it in the background if it stalls heartbeats